syntax = "proto3";

package idm;

option go_package = "idmService/proto";

service IdentityService {
  rpc Register(RegisterRequest) returns (RegisterResponse);
  rpc Login(LoginRequest) returns (LoginResponse);
  rpc VerifyToken(VerifyTokenRequest) returns (VerifyTokenResponse);
  rpc RevokeToken(RevokeTokenRequest) returns (RevokeTokenResponse);
}

message RegisterRequest {
  string email = 1;
  string password = 2;
  string role = 3;
}

message RegisterResponse {
  bool success = 1;
  string message = 2;
  string user_id = 3;
}

message LoginRequest {
  string email = 1;
  string password = 2;
}

message LoginResponse {
  bool success = 1;
  string token = 2;
  string message = 3;
  string user_id = 4;
  string role = 5;
  string email = 6;
}

message VerifyTokenRequest {
  string token = 1;
}

message VerifyTokenResponse {
  bool valid = 1;
  string email = 2;
  string message = 3;
  string user_id = 4;
  string role = 5;
  string issuer = 6;
  int64 expires_at = 7;
  bool expired = 8;
  bool blacklisted = 9;
}

message RevokeTokenRequest {
  string token = 1;
}

message RevokeTokenResponse {
  bool success = 1;
  string message = 2;
}
